src/julia_gc.c

```diff
@@ -744,17 +728,24 @@ jl_datatype_t * GAP_DeclareBag(jl_sym_t *      name,
                                1, large > 0);
 }
 
-#ifdef HAVE_JL_REINIT_FOREIGN_TYPE
-// internal wrapper for jl_boundp to deal with API change in Julia 1.12
-static int gap_jl_boundp(jl_module_t *m, jl_sym_t *var)
+#if defined(USE_GAP_INSIDE_JULIA)
+// for backwards compatibility, may be removed in the future
```

Comment:
But this is called by GAP.jl, why would we want to remove it?

Reply:
Note that there can be a *lot* of pain caused by such changes when bisecting issues in the C code -- when you change ABIs like that, it better be worth it, or the next time someone is trying to debug a GC weirdness they'll be cursing your name for this ;-).

Thus I'd really prefer to keep `GAP_InitJuliaMemoryInterface` even if we don't need the second argument anymore right now.